xsd-parser-rs/src/generator/struct.rs

```diff
@@ -35,17 +42,40 @@ pub trait StructGenerator {
     }
 
     fn subtypes(&self, entity: &Struct, gen: &Generator) -> String {
-        gen.base().join_subtypes(entity.subtypes.as_ref(), gen)
-    }
-
-    fn fields_subtypes(&self, entity: &Struct, gen: &Generator) -> String {
-        entity
+        let field_subtypes = entity
             .fields
             .borrow()
             .iter()
             .map(|f| gen.base().join_subtypes(f.subtypes.as_ref(), gen))
             .collect::<Vec<String>>()
-            .join("")
+            .join("");
+
+        let subtypes = gen.base().join_subtypes(entity.subtypes.as_ref(), gen);
+
+        if !field_subtypes.is_empty() || !subtypes.is_empty() {
+            format!(
+                "\npub mod {name} {{\n{indent}use super::*;{st}\n{fst}\n}}\n",
+                name = self.mod_name(entity, gen),
+                st = subtypes,
+                indent = gen.base().indent(),
+                fst = self.shift(&field_subtypes, gen.base().indent().as_str())
+            )
+        } else {
+            format!("{}\n{}", subtypes, field_subtypes)
+        }
+    }
+
+    fn shift(&self, text: &str, indent: &str) -> String {
+        text.replace("\n\n\n", "\n")
```

Comment:
This "\n\n\n" is weird. Add TODO here at least, I would like to figure out why we have so many redundant new lines.